src/Common/Config/YAMLParser.cpp

```diff
@@ -57,7 +57,15 @@ void processNode(const YAML::Node & node, Poco::XML::Element & parent_xml_elemen
     {
         case YAML::NodeType::Scalar:
         {
-            auto value = node.as<std::string>();
+            std::string value;
+            try
+            {
+                value = node.as<std::string>();
+            }
+            catch (const YAML::TypedBadConversion<std::string>&)
+            {
+                throw Exception(ErrorCodes::CANNOT_PARSE_YAML, "YAMLParser has encountered node with value which cannot be represented as string and cannot continue parsing of the file");
+            }
```

Comment:
You have repeated the same code three times. I think it's better to do that work only once and in another place - in the function parse() where you're already handling other exception types, like this:

```
    catch (const YAML::BadFile&)
    {
        /// yaml-cpp cannot open the file even though it exists
        throw Exception(ErrorCodes::CANNOT_OPEN_FILE, "Unable to open YAML configuration file {}", path);
    }
    catch (const YAML::TypedBadConversion<std::string>&)
    {
        throw Exception(ErrorCodes::CANNOT_PARSE_YAML, "YAMLParser has encountered node with value which cannot be represented as string and cannot continue parsing of the file: {}", path);
    }
    catch (const YAML::Exception& e)
    {
        /// yaml-cpp cannot parse the file because its contents are incorrect
        throw Exception(ErrorCodes::CANNOT_PARSE_YAML, "Unable to parse YAML configuration file {}", path, e.what());
    }
```